pkg/tfgen/generate_schema.go

```diff
@@ -908,3 +944,47 @@ func appendExample(description, markdownToAppend string) string {
 	}
 	return strings.Join(reassembledLines, "\n")
 }
+
+func modulePlacementForType(pkg tokens.Package, path *paths.TypePath) tokens.Module {
```

Comment:
This feels like the heart of the change. Lets add some unit tests here since none of the downstream repos actually exhibit the format that would trigger the bug in #611?